ui/components/app/assets/token-cell/token-cell.tsx

```diff
@@ -83,88 +74,33 @@ export default function TokenCell({
     return null;
   }
 
-  const PrimaryDisplay = () =>
-    TokenCellPrimaryDisplayOverride ? (
-      <TokenCellPrimaryDisplayOverride />
-    ) : (
-      <TokenCellPrimaryDisplay
-        token={{ ...token, ...tokenDisplayInfo }}
-        privacyMode={privacyMode}
-      />
-    );
-
   return (
-    <Box
-      display={Display.Flex}
-      flexDirection={FlexDirection.Row}
-      width={BlockSize.Full}
-      height={BlockSize.Full}
-      gap={4}
-    >
-      <Box
-        as="a"
-        onClick={(e?: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => {
-          e?.preventDefault();
-
-          if (!onClick || showScamWarningModal) {
-            return;
-          }
-
-          onClick();
-        }}
-        display={Display.Flex}
-        flexDirection={FlexDirection.Row}
-        paddingTop={2}
-        paddingBottom={2}
-        paddingLeft={4}
-        paddingRight={4}
-        width={BlockSize.Full}
-        style={{
-          height: 62,
-          cursor: onClick ? 'pointer' : 'auto',
-          backgroundColor:
-            !disableHover && isHovered
-              ? 'var(--color-background-default-hover)'
-              : 'transparent',
-          transition: 'background-color 0.2s ease-in-out',
-        }}
-        onMouseEnter={() => !disableHover && setIsHovered(true)}
-        onMouseLeave={() => !disableHover && setIsHovered(false)}
-        data-testid="multichain-token-list-button"
-      >
-        <TokenCellBadge token={{ ...token, ...tokenDisplayInfo }} />
-        <Box
-          display={Display.Flex}
-          flexDirection={FlexDirection.Column}
-          width={BlockSize.Full}
-          style={{ flexGrow: 1, overflow: 'hidden' }}
-          justifyContent={JustifyContent.center}
-        >
-          <Box
-            display={Display.Flex}
-            flexDirection={FlexDirection.Row}
-            justifyContent={JustifyContent.spaceBetween}
-          >
-            <TokenCellTitle token={{ ...token, ...tokenDisplayInfo }} />
-            <TokenCellSecondaryDisplay
-              token={{ ...token, ...tokenDisplayInfo }}
-              handleScamWarningModal={handleScamWarningModal}
-              privacyMode={privacyMode}
-            />
-          </Box>
-
-          <Box
-            display={Display.Flex}
-            flexDirection={FlexDirection.Row}
-            justifyContent={JustifyContent.spaceBetween}
-          >
-            <TokenCellPercentChange token={{ ...token, ...tokenDisplayInfo }} />
-            <PrimaryDisplay />
-          </Box>
-        </Box>
-      </Box>
-      {/* scam warning modal, this should be higher up in the component tree */}
-      {isEvm && showScamWarningModal ? (
+    <>
+      <GenericAssetCellLayout
+        onClick={showScamWarningModal ? undefined : onClick}
+        disableHover={disableHover}
+        badge={<AssetCellBadge {...{ ...token, ...tokenDisplayInfo }} />}
```

Comment:
I see that this was in the original logic, but yikes 🤣 

Can we move this into a variable since it is used in a few places?

```ts
const fullToken = useMemo(() => ({
  ...token,
  ...tokenDisplayInfo
}), [token, tokenDisplayInfo])
```